src/csharp/Microsoft.Spark/Sql/ArrowArrayHelpers.cs

```diff
@@ -113,57 +90,53 @@ private static ArrayData BuildEmptyArrayDataFromArrayType<T>()
-            }
-            else if (type == typeof(UInt64Array))
-            {
-                arrowType = UInt64Type.Default;
-            }
-            else if (type == typeof(FloatArray))
-            {
-                arrowType = FloatType.Default;
-            }
-            else if (type == typeof(DoubleArray))
-            {
-                arrowType = DoubleType.Default;
-            }
-            else if (type == typeof(Date64Array))
-            {
-                arrowType = Date64Type.Default;
-            }
-            else if (type == typeof(TimestampArray))
-            {
-                arrowType = TimestampType.Default;
+            switch (type)
+            {
+                case Type t when t == typeof(BooleanArray):
+                    arrowType = BooleanType.Default;
+                    break;
+                case Type t when t == typeof(Int8Array):
+                    arrowType = Int8Type.Default;
+                    break;
+                case Type t when t == typeof(UInt8Array):
+                    arrowType = UInt8Type.Default;
+                    break;
+                case Type t when t == typeof(Int16Array):
+                    arrowType = Int16Type.Default;
+                    break;
+                case Type t when t == typeof(UInt16Array):
+                    arrowType = UInt16Type.Default;
+                    break;
+                case Type t when t == typeof(Int32Array):
+                    arrowType = Int32Type.Default;
+                    break;
+                case Type t when t == typeof(UInt32Array):
+                    arrowType = UInt32Type.Default;
+                    break;
+                case Type t when t == typeof(Int64Array):
+                    arrowType = Int64Type.Default;
+                    break;
+                case Type t when t == typeof(UInt64Array):
+                    arrowType = UInt64Type.Default;
+                    break;
+                case Type t when t == typeof(FloatArray):
+                    arrowType = FloatType.Default;
+                    break;
+                case Type t when t == typeof(DoubleArray):
+                    arrowType = DoubleType.Default;
+                    break;
+                case Type t when t == typeof(Date64Array):
+                    arrowType = Date64Type.Default;
+                    break;
+                case Type t when t == typeof(TimestampArray):
+                    arrowType = TimestampType.Default;
+                    break;
+                case Type t when t == typeof(StringArray):
+                    return new ArrayData(StringType.Default, 0,
+                    buffers: new[] { ArrowBuffer.Empty, ArrowBuffer.Empty, ArrowBuffer.Empty });
+                case Type t when t == typeof(BinaryArray):
+                    return new ArrayData(BinaryType.Default, 0,
+                    buffers: new[] { ArrowBuffer.Empty, ArrowBuffer.Empty, ArrowBuffer.Empty });
```

Comment:
Similarly, I think we can do the following?
```csharp
        private static ArrayData BuildEmptyArrayDataFromArrayType<T>()
        {
            return typeof(T) switch
            {
                Type t when t == typeof(BooleanArray) =>
                    BuildEmptyArrayDataFromArrowType(BooleanType.Default),
                Type t when t == typeof(Int8Array) =>
                    BuildEmptyArrayDataFromArrowType(Int8Type.Default),
                ...
                Type t when t == typeof(StringArray) =>
                    BuildEmptyArrayDataFromArrowType(StringType.Default),
                ...
                _ => throw new NotSupportedException($"Unknown type:  {typeof(T)}")
            };
        }
```

Reply:
I think it looks better!
I'll fix and commit again.

Thanks @imback82

Reply:
I reflected your reviews!